sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -78,7 +80,66 @@ private string GetSessionFilePath()
             string className = testAdapter.ClassName.Substring(testAdapter.ClassName.LastIndexOf('.') + 1);
 
             string fileName = name + (IsAsync ? "Async" : string.Empty) + ".json";
-            return Path.Combine(TestContext.CurrentContext.TestDirectory,
+
+            var path = TestContext.CurrentContext.TestDirectory;
+            var indexartifacts = path.IndexOf("artifacts");
+            if (indexartifacts == -1)
+            {
+                return Path.Combine(TestContext.CurrentContext.TestDirectory,
+                "SessionRecords",
+                additionalParameterName == null ? className : $"{className}({additionalParameterName})",
+                fileName);
+            }
+            var newpath = path.Substring(0, indexartifacts) + "sdk";
+
+            // check if Track 1 or Track 2
+            int firstIndex = path.IndexOf("bin\\");
+            int lastIndex = path.LastIndexOf(".Tests");
+            if (firstIndex == -1 || lastIndex == -1)
+            {
+                return Path.Combine(TestContext.CurrentContext.TestDirectory,
+                "SessionRecords",
+                additionalParameterName == null ? className : $"{className}({additionalParameterName})",
+                fileName);
+            }
+
+            firstIndex += "bin\\".Length;
+            var testname = path.Substring(firstIndex, lastIndex - firstIndex);
+            var rpname = "";
+
+            if (testname.Substring(0,5) == "Azure") //Track 2
+            {
+                int rpnamefirstindex = testname.IndexOf("ResourceManager.") + "ResourceManager.".Length;
+                rpname = testname.Substring(rpnamefirstindex, testname.Length - rpnamefirstindex).ToLower();
```

Comment:
Need to test if it management. Code should be if check if Azure.ResourceManager.